src/hooks/useLogin.ts

```diff
@@ -0,0 +1,67 @@
+import { useContext, useEffect, useState } from "react";
+import { FirebaseError } from "firebase/app";
+import {
+	getAuth,
+	onAuthStateChanged,
+	signInWithEmailAndPassword,
+} from "firebase/auth";
+
+import { db, firebaseApp } from "../config/firebase";
+import { AuthContext } from "../context/AuthContext";
+import { doc, getDoc } from "firebase/firestore";
+
+type UserData = {
+	email: string;
+	password: string;
+};
+
+export const useLogin = () => {
+	const [error, setError] = useState<string | null>(null);
+	const [isPending, setIsPending] = useState<boolean>(false);
+	const context = useContext(AuthContext);
+
+	const auth = getAuth(firebaseApp);
+	const loginUser = async (data: UserData) => {
+		setIsPending(true);
+		setError(null);
+		try {
+			const res = await signInWithEmailAndPassword(
+				auth,
+				data.email,
+				data.password,
+			);
+			if (!res) throw new Error("Some error during register.");
+			setError(null);
+		} catch (error) {
+			if (error instanceof FirebaseError) {
+				console.log(error.code);
+				if (error.code.includes("invalid-credential")) {
+					return setError("The credential used to authenticate are not valid.");
+				}
+
+				setError("Some error during loging. Try again later.");
+			}
+		} finally {
+			setIsPending(false);
+		}
+	};
+
+	useEffect(() => {
+		onAuthStateChanged(auth, async (user) => {
+			if (user) {
+				const docRef = doc(db, "User", user.uid);
+				const docSnap = await getDoc(docRef);
+				if (docSnap.exists())
+					context?.dispatch({ type: "LOGIN", payload: docSnap.data() });
```

Comment:
id usera masz zapisane w kolekcji Users? Jeżeli nie to zmień tą linijkę na:
```
context?.dispatch({ type: "LOGIN", payload: {...docSnap.data(), id: user.uid });
```

Reply:
Tak, w czasie rejestracji dodaje pole id